### PR TITLE
feat: added test suites

### LINKED ISSUE #22542: Create functional test suite in E2E typescript tests
### Is your task related to a problem? Please describe

It's needed to run all functional tests from https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/complex/job/latest/job/functional-tests/ in one suite:
- CreateWorkspaceWithExistedName
- Quarkus
- EmptyWorkspace
- ContainerOverridesAPI
- PodOverridesAPI
- "Java 11 with Quarkus" RecommendedExtensions test
- "Java 11 with Lombok" RecommendedExtensions test
- "Python" RecommendedExtensions test
- "Go" RecommendedExtensions test
- "PHP" RecommendedExtensions test
- "Node.js Express" RecommendedExtensions test
- "Node.js MongoDB" RecommendedExtensions test
- "C/C++" RecommendedExtensions test
- ".NET" RecommendedExtensions test
- "Ansible" RecommendedExtensions test
- SmokeTest
- PredefinedNamespace
- DevConsoleIntegration
- DevfileAcceptanceTestAPI
- WorkspaceWithParent
- Documentation
- (PodmanTest)

Tests should be executed one by one, and suite run should not be aborted because of test failure.

### Describe the solution you'd like

-

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nallikaea**
### What does this PR do?
Closes #22542 

### Screenshot/screencast of this PR

NA

### What issues does this PR fix or reference?

#22542 

### How to test this PR?

set urls for testing application and passwords than run `tests/e2e/configs/sh-scripts/regressionTestRun.sh`

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by musienko-maxim — APPROVED** [review]

**Comment by nallikaea** [comment]
> @nallikaea: is it possible to trigger functional test suite by using single environment variable, e.g. `SUITE=functional-tests`?

an idea set architecture  and run `tests/e2e/configs/sh-scripts/regressionTestRun.sh` that is it. but now it is only for x86_64

**Comment by dmytro-ndp** [comment]
well, an idea behind #22542 was to have separate single suite for functional tests, not three ones

**Comment by nallikaea** [comment]
> well, an idea behind #22542 was to have separate single suite for functional tests, not three ones

I divided it in next PR https://github.com/eclipse/che/pull/22590/commits/56b9477b4175c420ba260b0697a732762eb0d88e#diff-91a34f7999689e2ba18658f8a1527cc09e08603a9bf0915f16771c7017cc689b

**Comment by dmytro-ndp** [comment]
@nallikaea: could you please consider unified solution for both x86_64, IBM Z/Power architectures, which doesn't depend on OCP version, with simple entry point `SUITE=functional-tests` ?

**Comment by nallikaea** [comment]
@dmytro-ndp check new approach according our discussion, please

File: tests/e2e/constants/BASE_TEST_CONSTANTS.ts [hunk 1]
```diff
@@ -48,6 +67,9 @@ export const BASE_TEST_CONSTANTS: {
 	 * is cluster disconnected of online
 	 */
 	IS_CLUSTER_DISCONNECTED: (): boolean => BASE_TEST_CONSTANTS.TS_SELENIUM_BASE_URL.includes('airgap'),
+	OCP_TYPE: (): string => {
```
Comment (dmytro-ndp) [inline tests/e2e/constants/BASE_TEST_CONSTANTS.ts]: Why not to continue using `IS_CLUSTER_DISCONNECTED` constant?

Comment (nallikaea) [reply tests/e2e/constants/BASE_TEST_CONSTANTS.ts]: because it is returns us boolean but I need string

**Comment by nallikaea** [comment]
> It's also needed to know how to call `regressionTestRun.sh` when run quay.io/eclipse/che-e2e:next image container.
@dmytro-ndp there is the script in package.json 
The command should be
```
docker run --shm-size=4096m -p 5920:5920 \
  -e TS_SELENIUM_BASE_URL=<url> \
  -e 'TS_SELENIUM_OCP_PASSWORD=<pass>' \
  -e OCP_VERSION="4.13" \
  - e ARCH_VERSION="x86_64" \
  -v /tmp:/usr/local/bin:Z \
  -v /home/hudson/.kube:/home/seluser/.kube:Z \
  -v test-repo/tests/e2e/report:/tmp/e2e/report:Z \
  -v test-repo/tests/e2e/video:/tmp/ffmpeg_report:Z \
  -e TEST_SUITE=functional-test-suite \
  ' quay.io/eclipse/che-e2e:next
```

**Comment by nallikaea** [comment]
fixed typo suits

**Comment by nallikaea** [comment]
https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/basic/job/typescript-tests/13184/

File: tests/e2e/configs/sh-scripts/runFunctionalTests.sh [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  echo ""
+  echo "Launching all tests for $TEST_ENVIRONMENT"
+  echo ""
+  launchDynamicallyGeneratingAPITests
+  launchAPITests
+  launchUITests
+}
+
+initTestValues() {
+  if [[ "$TS_SELENIUM_BASE_URL" =~ "airgap" ]]
```
Comment (dmytro-ndp) [inline tests/e2e/configs/sh-scripts/runFunctionalTests.sh]: We have dedicated `IS_CLUSTER_DISCONNECTED` constant aimed to declare disconnected cluster.

Also, IBM Z / Power airgap OCP clusters doesn't contain `airgap` prefix in their host addresses.

So, what do you think about next condition which is checking that IS_CLUSTER_DISCONNECTED is defined and true?
```
if [[ -n "$IS_CLUSTER_DISCONNECTED" && "$IS_CLUSTER_DISCONNECTED" == "true" ]]
```

File: tests/e2e/constants/BASE_TEST_CONSTANTS.ts (hunk 1 above)
Comment (dmytro-ndp) [reply tests/e2e/constants/BASE_TEST_CONSTANTS.ts]: I meant, `OCP_TYPE` is actually duplicating `IS_CLUSTER_DISCONNECTED`.

File: tests/e2e/configs/sh-scripts/runFunctionalTests.sh (hunk 2 above)
Comment (nallikaea) [reply tests/e2e/configs/sh-scripts/runFunctionalTests.sh]: how about ?
```
initTestValues() {
  if [[ "$TS_SELENIUM_BASE_URL" =~ "airgap" || (-n "$IS_CLUSTER_DISCONNECTED" && "$IS_CLUSTER_DISCONNECTED" == "true") ]]
  then
     echo "Disconnected environment"
     export OCP_TYPE="disconnected-ocp"
  else
     echo "Online environment"
     export OCP_TYPE="online-ocp"
  fi
  ```

File: tests/e2e/constants/BASE_TEST_CONSTANTS.ts (hunk 1 above)
Comment (nallikaea) [reply tests/e2e/constants/BASE_TEST_CONSTANTS.ts]: I can`t get why
```
IS_CLUSTER_DISCONNECTED: (): boolean => BASE_TEST_CONSTANTS.TS_SELENIUM_BASE_URL.includes('airgap'),
// returns boolean true or false
OCP_TYPE: (): string => {
		return process.env.OCP_TYPE || (BASE_TEST_CONSTANTS.IS_CLUSTER_DISCONNECTED() ? 'disconnected-ocp' : 'online-ocp');
// returns  'disconnected-ocp' or 'online-ocp' and uses as name directory with suit file
	},
```
you don`t need to set OCP_TYPE it automatically defines, but you can. How should I rewrite it on your opinion? Maybe I missed something =)

Comment (dmytro-ndp) [reply tests/e2e/constants/BASE_TEST_CONSTANTS.ts]: My opinion is that it should be enough to have `IS_CLUSTER_DISCONNECTED` only in constant list.
Duplicating constant `OCP_TYPE` is rather misleading than helpful.

File: tests/e2e/configs/sh-scripts/runFunctionalTests.sh (hunk 2 above)
Comment (dmytro-ndp) [reply tests/e2e/configs/sh-scripts/runFunctionalTests.sh]: okay, looks good

**Review by dmytro-ndp — APPROVED** [review]
Thanks for taking review remarks into account.

IMHO looks good to merge.

**Comment by dmytro-ndp** [comment]
@nallikaea : let me try to run the test suite from CRW Jenkins CI before the merge this PR.

**Comment by nallikaea** [comment]
> @nallikaea : let me try to run the test suite from CRW Jenkins CI before the merge this PR. Update: https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/basic/job/typescript-tests/13308/
> 
> Update 2: tests has finished execution. I had found few tests failed on workspace stop test. PR is ready to merge.

If we are running full tests suite we should not stop our test run after first test fail. In this case we have mocha bail parameter as false [1]. On the other hand it will create a conflict between tests `test('Stop the workspace', async function () {...})` and method  `function deleteWorkspaceOnFailedTest(this: Mocha.Context)` which is triggering on test fail if `DELETE_WORKSPACE_ON_FAILED_TEST=true` 
There is task[2] already created to simplify this behavior.

[1] [Make it possible not to interrupt test suite execution after separate test has failed](https://issues.redhat.com/browse/CRW-582)
[2] [Change Login, Logout, Stop and delete workspace tests to be hooks (suileSetup, suiteTeardown) #22587](https://github.com/eclipse/che/issues/22587)

**Pull request merged**